src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetGroupName.cs

```diff
@@ -7,17 +7,23 @@
 using System;
 using System.Collections.Generic;
 using System.Reflection;
+using System.IO;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
         /// <summary>
-        /// Gets the group name associated to the specified group ID.
+        /// Tries to get the group name associated to the specified group ID.
         /// </summary>
         /// <param name="gid">The group ID.</param>
-        /// <returns>On success, return a string with the group name. On failure, throws an IOException.</returns>
-        internal static string GetGroupName(uint gid) => GetGroupNameInternal(gid) ?? throw GetIOException(GetLastErrorInfo());
+        /// <param name="groupName">When this method returns true, gets the value of the group name associated with the specified id. On failure, it is null.</param>
+        /// <returns>On success, return true. On failure, returns false.</returns>
+        internal static bool TryGetGroupName(uint gid, out string? groupName)
```

Comment:
nit: we can hit the compiler that it's not a null when the method returns true

```suggestion
        internal static bool TryGetGroupName(uint gid, [NotNullWhen(true)] out string? groupName)
```

Reply:
Good point. Although "hit the compiler" sounds too violent, I'd rather "hint the compiler" 😄.